메인 페이지 인기앨범차트 디자인

kkwonwon opened the pull request:
## #️⃣ 연관이슈 

- close #22 

## 📝 작업 내용

메인 페이지의 인기 앨범 차트 디자인입니다.

차트 부분은 왼쪽과 오른쪽을 나누어 AlbumList함수에서 인덱스를 받아 쓰는 방식으로 만들었습니다.

## 의논할 거리
별점과 재사용 가능한 부분은 따로 나누면 좋지 않을까 회의에서 이야기 나누었었고
개인적으론 해당 코드에서 별점의 색을 Hex코드로 쓰고 있어 아직은 가독성이 떨어지는게 흠인듯 합니다.
그리고 폰트 크기나 사이즈, 마진을 다른데에서 가져오지않고 직접 써놓아서 만약 고칠 부분이 보이신다면 피드백 부탁드립니다!

kkwonwon committed:
별점 기능외 디자인 추가

kkwonwon committed:
인기 앨범 차트 별점 추가

hangooksaram commented on src/components/AlbumChart.tsx:
```diff
@@ -1,8 +1,108 @@
+        gridTemplateRows: 'repeat(4, 1fr)',
+      }}
+    >
+      {Array.isArray(data?.top5Albums) ? (
```
`top5Albums` 가 `Array` 타입 외의 타입이 될 가능성이 있는지 확인해보면 좋을 것 같습니다.

kkwonwon replied:
일단 `Array` 타입 이외로 `top5Albums`가 들어오면 `<Typography>List Empty</Typography>` 띄우도록 처리하고 있긴 한데 어떤 방법이 좋을까요?

hangooksaram replied:
저 코멘트는 `top5Albums` 가 `Array` 타입 외의 타입이 될 가능성이 없다면, `Array.isArray` 와 같은 메서드로 타입 체크할 필요가 없기 때문에 그 부분을 확인 해보면 좋을것 같아서 말씀드린 거였어요!

다른 코드에서도 타입이 `Array` 로 고정되어있는데, `Array.isArray` 메서드로 타입 체크를 하는 부분이 있어 확인해보면 좋을 것 같아 말씀드린 거였습니다.

kkwonwon replied:
`!data?.top5Albums` 와 같이하면 List Empty 가 떠서 일단 해당 방법으로 머지하고 차후에 다른 코드들도 한번에 `Array.isArray` 부분을 바꾸는 건 어떨까요?

hangooksaram commented on src/components/AlbumChart.tsx:
```diff
@@ -1,8 +1,108 @@
+                }}
+              />
+              <CardContent sx={{ marginTop: '8px' }}>
+                <Typography align="left" sx={{ fontSize: '14px' }}>
```
Typography 의 fontSize 속성에 값을 직접 지정할 수 있습니다. `fontSize = "fontSizeMd"`

hangooksaram commented:
_말씀하신 저희가 `mui` 의 `theme` 을 통해 지정해둔 폰트 크기 와 같은 값들을 문자열로 가져와서 쓰다보니 확인이 어려운 부분이있어 `theme` 객체에 접근 가능하도록 수정하는 게 좋을 것 같습니다!_

-> 이 부분은 현재 `theme` 이 자바스크립트 로 작성되어 내부 속성이 편집기에서 보이지 않았을 뿐, `useTheme` 이라는 훅을 통해 `theme` 객체에 접근이 가능하다고 합니다!

그리고 컬러값 또한 상수화하여 관리하면 좋을 듯 하네요!

kkwonwon commented:
피드백 주신대로 수정해서 다시 커밋하겠습니다!

kkwonwon committed:
theme수정 fontSize사용

theme 의 palette 에 star 색을 추가
fontSize 사용해서 폰트크기 수정
useTheme은 일단 사용하지 않아서 주석처리
카드 클릭 시 /album/:id 쪽으로 이동

kkwonwon committed:
EOL auto

chyon8 commented:
머지해주시면 될것같습니다~
